gem/apps/gem/routes/users.py

```diff
@@ -25,63 +24,57 @@ def should_be_at_least_6_chars_long(cls, v: str):
 
 
 @router.post("/")
-async def create_user(user: User) -> models.User:
-    with session_scope() as s:
-        # s.expire_on_commit = False
-        user_db = map_user_to_model(user)
-        user_db.hashed_password = pwd_context.hash(user.password)
-        s.add(user_db)
-        s.flush()
-        user.oid = user_db.id
-        return user
+async def create_user(user: User, s: Session = Depends(get_db)) -> models.User:
+    # s.expire_on_commit = False
+    user_db = map_user_to_model(user)
+    user_db.hashed_password = pwd_context.hash(user.password)
+    s.add(user_db)
+    s.flush()
+    user.oid = user_db.id
+    return user
 
 
 @router.put("/")
-async def update_user(user: User):
-    with session_scope() as s:
-        user_db = s.query(models.User).filter_by(
-            id=user.oid).first()  # type: models.User
-        if not user_db:
-            return False
-        user_db.username = user.username
-        user_db.full_name = user.full_name
-        user_db.email = user.email
-        user_db.disabled = user.disabled
-        s.commit()
+async def update_user(user: User, s: Session = Depends(get_db)):
+    user_db = s.query(models.User).filter_by(id=user.oid).first()  # type: models.User
+    if not user_db:
+        return False
```

Comment:
```suggestion
        raise HTTPException(status_code=404, detail="User not found")
```
Тут тоже надо вернуть 404 :)